File: apps/askbible-android/app/src/main/java/me/askbible/native_/ui/HighlightBar.kt
package me.askbible.native_.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.navigationBarsPadding
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import me.askbible.native_.data.AppLocale
import me.askbible.native_.data.Parchment
import me.askbible.native_.data.SiteCopy
import me.askbible.native_.data.VerseHighlightRules

/**
 * 划重点时底部的调色卡片：四个颜色 + 擦除 + 完成。
 * 和长按操作单同一张羊皮卡片；由壳画在播放坞之上。与 iOS 的 HighlightBar 对等。
 */
@Composable
fun HighlightBar(
    locale: AppLocale,
    color: String,
    erasing: Boolean,
    onPickColor: (String) -> Unit,
    onErase: () -> Unit,
    onDone: () -> Unit,
    theme: Parchment = Parchment.light,
) {
    Box(Modifier.fillMaxSize(), contentAlignment = Alignment.BottomCenter) {
        Column(
            Modifier.fillMaxWidth().navigationBarsPadding().parchmentCard(16.dp)
                .padding(start = 18.dp, end = 18.dp, top = 14.dp, bottom = 28.dp).clickableNoRipple {},
        ) {
            Row(Modifier.fillMaxWidth().padding(bottom = 12.dp), verticalAlignment = Alignment.Top) {
                Column(Modifier.weight(1f)) {
                    Text(SiteCopy.t("native.highlight", locale),
                         color = theme.ink.toColor(), fontSize = 17.sp, fontWeight = FontWeight.SemiBold)
                    Text(SiteCopy.t("native.highlightHint", locale),
                         color = theme.muted.toColor(), fontSize = 12.sp)
                }
                Text(SiteCopy.t("native.done", locale), Modifier.clickableNoRipple(onDone),
                     color = theme.ink.toColor(), fontSize = 15.sp, fontWeight = FontWeight.SemiBold)
            }
            Row(Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                for (hex in VerseHighlightRules.PALETTE) {
                    val on = !erasing && hex == color
                    Box(
                        Modifier.size(34.dp).clip(CircleShape).background(hexColor(hex))
                            .border(if (on) 2.dp else 0.5.dp,
                                    theme.ink.toColor().copy(alpha = if (on) 0.75f else 0.12f), CircleShape)
                            .clickableNoRipple { onPickColor(hex) },
                    )
                }
                Spacer(Modifier.weight(1f))
                Box(
                    Modifier.heightIn(min = 34.dp).clip(RoundedCornerShape(999.dp))
                        .background(if (erasing) Color(0x38FFB101) else Color(0x80FFF8EB))
                        .clickableNoRipple(onErase).padding(horizontal = 12.dp),
                    contentAlignment = Alignment.Center,
                ) {
                    Text(SiteCopy.t("native.highlightErase", locale),
                         color = if (erasing) theme.ink.toColor() else theme.muted.toColor(),
                         fontSize = 13.sp, fontWeight = FontWeight.SemiBold)
                }
            }
        }
    }
}

File: apps/askbible-ios/AskBible/Read/HighlightBar.swift
import SwiftUI

/// 划重点时底部的调色卡片：四个颜色 + 擦除 + 完成。
/// 和长按操作单同一张羊皮卡片；画在播放坞之上，否则会被坞挡住。
struct HighlightBar: View {
    let locale: AppLocale
    @Binding var color: String
    @Binding var erasing: Bool
    var onDone: () -> Void

    @Environment(\.parchment) private var theme

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            HStack(alignment: .top, spacing: 12) {
                VStack(alignment: .leading, spacing: 2) {
                    Text(SiteCopy.t("native.highlight", locale))
                        .font(.system(size: 17, weight: .semibold))
                        .foregroundStyle(theme.ink)
                    Text(SiteCopy.t("native.highlightHint", locale))
                        .font(.system(size: 12))
                        .foregroundStyle(theme.muted)
                }
                .frame(maxWidth: .infinity, alignment: .leading)
                Button(action: onDone) {
                    Text(SiteCopy.t("native.done", locale))
                        .font(.system(size: 15, weight: .semibold))
                        .foregroundStyle(theme.ink)
                        .contentShape(Rectangle())
                }
                .buttonStyle(.plain)
            }
            .padding(.bottom, 12)

            HStack(spacing: 12) {
                ForEach(VerseHighlightRules.palette, id: \.self) { hex in
                    Button {
                        color = hex
                        erasing = false
                    } label: {
                        // 所见即所得：色块用正文里实际铺出来的合成色，不是纯颜料色
                        Circle()
                            .fill(theme.canvas)
                            .overlay(Circle().fill(Color(uiColor: VerseHighlightRules.fillColor(hex))))
                            .frame(width: 34, height: 34)
                            .overlay(
                                Circle().stroke(theme.ink.opacity(!erasing && color == hex ? 0.75 : 0.12),
                                                lineWidth: !erasing && color == hex ? 2 : 0.5)
                            )
                            .contentShape(Circle())
                    }
                    .buttonStyle(.plain)
                }
                Spacer(minLength: 8)
                Button { erasing = true } label: {
                    Text(SiteCopy.t("native.highlightErase", locale))
                        .font(.system(size: 13, weight: .semibold))
                        .foregroundStyle(erasing ? theme.ink : theme.muted)
                        .padding(.horizontal, 12).frame(minHeight: 34)
                        .background(
                            Capsule().fill(erasing ? Color(rgb: 0xffb101, opacity: 0.22) : Color(parchment: 0xfff8eb, opacity: 0.5))
                        )
                        .contentShape(Rectangle())
                }
                .buttonStyle(.plain)
            }
        }
        .padding(.horizontal, 18).padding(.top, 14).padding(.bottom, 28)
        .frame(maxWidth: .infinity, alignment: .leading)
        // Josh 2026-09-16 真机确认画笔正常后改成玻璃。玻璃背景不像原来那张整屏羊皮图会吃触点
        // （DECISIONS 2026-09-13 Bug 2），所以不再需要 backgroundHitTesting: false 那个补丁。
        .askGlassRect(tone: .light, radius: AskCorner.card)
        .askFloatingShadow(.overlay)
        .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .bottom)
        .ignoresSafeArea(edges: .bottom)
    }
}
